rpc/flipt/evaluation/evaluation.proto

```diff
@@ -0,0 +1,72 @@
+syntax = "proto3";
+
+package flipt.evaluation;
+
+import "flipt.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "go.flipt.io/flipt/rpc/flipt/evaluation";
+
+message EvaluationRequest {
+  string request_id = 1;
+  string namespace_key = 2;
+  string flag_key = 3;
+  string entity_id = 4;
+  map<string, string> context = 5;
+}
+
+message BatchEvaluationRequest {
+  string request_id = 1;
+  repeated EvaluationRequest requests = 3;
+  bool exclude_not_found = 4;
+}
+
+message BatchEvaluationResponse {
+  string request_id = 1;
+  repeated EvaluationResponse responses = 2;
+  double request_duration_millis = 3;
+}
+
+enum EvaluationReason {
+    UNKNOWN_EVALUATION_REASON = 0;
+    FLAG_DISABLED_EVALUATION_REASON = 1;
+    FLAG_NOT_FOUND_EVALUATION_REASON = 2;
+    MATCH_EVALUATION_REASON = 3;
+    ERROR_EVALUATION_REASON = 4;
+    DEFAULT_EVALUATION_REASON = 5;
+}
+
+message EvaluationResponse {
+    oneof response {
+        BooleanEvaluationResponse boolean_response = 1;
+        VariantEvaluationResponse variant_response = 2;
+    }
+}
+
+message BooleanEvaluationResponse {
+  bool value = 1;
+  oneof rule {
+    RolloutSegment segment = 10;
+    RolloutPercentage percentage = 11;
+  }
```

Comment:
@markphelps while planning we were just discussing how important it is to add this and whether or not it is an implementation detail useful for testing, but not useful for end users.

Am I right in thinking: we added this so that a client could know which rule led to the match?
Do we forsee folks needed this from the outset, or should we drop it until we have a match?

If we keep it, is it possible to make a oneof `optional`?

Reply:
@GeorgeMac @markphelps Nice!